Document/0x04f-Testing-Network-Communication.md

```diff
@@ -205,67 +205,123 @@ If you want to verify whether your server supports the right cipher suites, ther
+You just have to hook the right functions, e.g., `SSL_write` and `SSL_read` from OpenSSL.
+
+This would work pretty well for apps using standard API library functions and classes; however, there might be some downsides:
+
+- The app might implement a custom network stack and you'll have to spend time analyzing the app to find out the APIs that you can use. See section "Searching for OpenSSL traces with signature analysis" in [this blog post](https://hackmag.com/security/ssl-sniffing/ "Searching for OpenSSL traces with signature analysis").
+- It might be very time consuming to craft the right hooking scripts to re-assemble HTTP response pairs (across many method calls and execution threads). You might find [ready-made scripts](https://github.com/fanxs-t/Android-SSL_read-write-Hook/blob/master/frida-hook.py) and even for [alternative network stacks](https://codeshare.frida.re/@owen800q/okhttp3-interceptor/), but depending on the app and the platform, these scripts might need a lot of maintenance and might not _always work_.
 
 See some examples:
 
 - ["Universal interception. How to bypass SSL Pinning and monitor traffic of any application"](https://hackmag.com/security/ssl-sniffing/), sections "Grabbing payload prior to transmission" and "Grabbing payload prior to encryption"
 - ["Frida as an Alternative to Network Tracing"](https://gaiaslastlaugh.medium.com/frida-as-an-alternative-to-network-tracing-5173cfbd7a0b)
 
-> This technique is also useful for other types of traffic such as BLE, NFC, etc. where deploying a MITM attack might be very costly and or complex.
+## MASTG-TECH: Passive Eavesdropping
+
+This method involves passively capturing network traffic using tools such as Wireshark or tcpdump (@MASTG-TOOL-0081, @MASTG-TOOL-0080, or @MASTG-TOOL-0075). It is useful for identifying network endpoints, analyzing protocol metadata, and understanding how an app communicates with its server. However, it cannot automatically decrypt TLS-encrypted communication. That said, [TLS decryption is possible](https://wiki.wireshark.org/TLS#tls-decryption) if you can [obtain the pre-master secret](https://wiki.wireshark.org/TLS#using-the-pre-master-secret). For an example specific to Android, see [this article](https://nibarius.github.io/learning-frida/2022/05/21/sniffing-tls-traffic).
+
+### When Is it Useful?
+
+Passive eavesdropping is particularly useful in the following scenarios:
+
+- When mobile apps use cross-platform frameworks such as [Xamarin](https://www.xamarin.com/platform "Xamarin") that ignore system proxy settings.
+- When mobile apps detect and prevent the use of system proxies.
+- When intercepting push notifications, such as Google Cloud Messaging (GCM)/Firebase Cloud Messaging (FCM) on Android.
+- When monitoring XMPP or other non-HTTP-based communication protocols.
+
+### How Does It Work?
+
+Route the traffic from the mobile device to the host computer:
```

Comment:
Passive collection is possible on a rooted Android and iOS device with tcpdump (so no need for host computer).